QueryBuilder/Typed/SelectQuery.cs

```diff
@@ -35,20 +36,37 @@ public TQuery Top(ushort numberOfRecords)
     /// The query that has the default select that was generated by the FROM clause.
     /// </summary>
     public class DefaultSelectQuery<T> : SelectQueryBase<DefaultSelectQuery<T>>
-        where T : BasicDigitalTwin
     {
         internal DefaultSelectQuery(IDictionary<string, Type> aliasToTypeMapping, SelectClause selectClause, FromClause fromClause, IList<JoinClause> joinClauses, WhereClause whereClauses) : base(aliasToTypeMapping, selectClause, fromClause, joinClauses, whereClauses)
         {
         }
 
+        /// <summary>
+        /// Select a property on a type and optionally provide an alias.
+        /// </summary>
+        /// <typeparam name="TSelect">The type from which to select a property.</typeparam>
+        /// <typeparam name="TOut">The type of the property.</typeparam>
+        /// <param name="propertySelector">The property to select on the type.</param>
+        /// <param name="propertyAlias">Optional alias to map to the selected property.</param>
+        /// <param name="typeAlias">Optional alias to map to the selected type.</param>
+        /// <returns>ADT query instance with one select clause.</returns>
+        public Query<TOut> Select<TSelect, TOut>(Expression<Func<TSelect, TOut>> propertySelector, string propertyAlias = null, string typeAlias = null)
```

Comment:
Couldn't we still maintain the `where TSelect : BasicDigitalTwin` constraint? I get that we'd need to remove this constraint on the class level, but on the individual methods, I feel like we should be able to keep this to enforce some semblance of needing the ADT SDK models. The only thing I imagine might be a hang up, is I know the generic type constraint can be weird about having a method overload that's only differentiated by a different type constraint, so if having one `Select<TSelect... where T : BasicDigitalTwin` and another be `Select<TSelect... where T : BasicRelationship` doesn't work, then just ignore this comment.

Reply:
Indeed, the compiler doesn't allow that. When I attempt to add a second version for `BasicRelationship` this error is thrown:
"CS0111: Type 'DefaultSelectQuery<T>' already defines a member called 'Select' with the same parameter types"

Reply:
Darn compiler doesn't know what's good for it... :(